node/resolvers/index.ts

```diff
@@ -1389,7 +1389,9 @@ export const resolvers = {
       }
 
       if (search) {
-        whereArray.push(`name="*${search}*"`)
+        whereArray.push(
+          `(name="*${search}*" OR b2bCustomerAdmin.email=${search})`
+        )
```

Comment:
You could validate the search string to check if there is an email pattern like email@domain.com and then add it to the query b2bCustomerAdmin.email or you have to change the pattern to email="\*${search}\*".
```ts
const emailExistsClause = search.match(/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?
/) ? " OR b2bCustomerAdmin.email=${search}" : "";

whereArray.push(
          `(name="*${search}*"  ${emailExistsClause})`
        )
```

Regex: [https://regex101.com/r/857lzc/1](https://regex101.com/r/857lzc/1)